azimuth/utils/ml/perturbation_functions.py

```diff
@@ -139,6 +140,7 @@ def remove_or_add_final_punctuation(
     # Replace with specified punctuation if last character is another punctuation sign.
     elif any([p in last_char for p in [".", "!", "?", ","]]):
         perturb_utt = original[:-1] + punctuation_sign
+        perturb_utt = re.sub(r" ([.!?,])", r"\1", perturb_utt)  # Remove any penultimate space
```

Comment:
I think we can simply remove the trailing space from `original[:-1]` with `rstrip()`.
```suggestion
        perturb_utt = original[:-1].rstrip() + punctuation_sign
```